File: frontend/src/types/DataModel.ts
export enum FieldType {
  STRING = 'STRING',
  NUMBER = 'NUMBER',
  BOOLEAN = 'BOOLEAN',
  ENUM = 'ENUM',
}

export enum DataModelCategory {
  INPUT = 'INPUT',
  OUTPUT = 'OUTPUT',
  INTERNAL = 'INTERNAL',
}

export interface FieldDefinition {
  name: string;
  type: FieldType;
  enumName?: string;
}

export interface DataModel {
  name: string;
  description: string;
  category?: DataModelCategory;
  source?: string;
  fields: FieldDefinition[];
}

export interface EnumDefinition {
  name: string;
  description: string;
  values: string[];
}

File: src/main/java/com/demo/common/DataModel.java
package com.demo.common;

import lombok.Data;

import java.util.List;

@Data
public class DataModel {
    private String name;
    private String description;
    private DataModelCategory category;
    private DataModelSourceType sourceType;
    private String source;
    private List<FieldDefinition> fields;
}
